app/email.go

```diff
@@ -295,7 +295,7 @@ func (es *EmailService) SendCloudWelcomeEmail(userEmail, locale, teamInviteID, w
 	T := i18n.GetUserTranslations(locale)
 	subject := T("api.templates.cloud_welcome_email.subject")
 
-	workSpacePath := fmt.Sprintf("https://%s.cloud.mattermost.com", workSpaceName)
+	workSpacePath := fmt.Sprintf("https://%s", dns)
```

Comment:
It might be better to use the SiteURL config variable for this as opposed to the DNS name. It's what the rest of the emails (even the cloud ones) use. That way, you don't have to mess with any string formatting.